block-library/html/index.js

```diff
@@ -30,6 +30,7 @@ export const settings = {
 		customClassName: false,
 		className: false,
 		html: false,
+		htmlValidation: false,
```

Comment:
I'm a little bit worried about this flag, people could use it to upgrade their blocks without worrying about deprecated versions. I'd be more inclined to look at what we're doing wrong when comparing the saved and parsed HTML

Reply:
By people do you mean developers or writers? I’m very unfamiliar with upgrading blocks and why it might be a problem.

I agree that improving Gutenberg’s handling of HTML will help. I'm going to try and see what can be done separately, although I suspect it will always be incomplete/imperfect.

My understanding (which could be very wrong!) is the custom HTML block should allow any HTML, regardless of correctness, and should leave it alone - no validation, formatting, or anything other than sanitising for my user level. It’s the block I pick if I want Gutenberg to stay out of my (non-block) stuff.

Would you be less worried if this was limited to `core/html`? Or is that it has to be disabled at all?

Reply:
I meant developers which could create issues for users upgrading to the latest version of these blocks (Content loss).

> My understanding (which could be very wrong!) is the custom HTML block should allow any HTML, regardless of correctness, and should leave it alone

This part I agree with and the fact that the block is not idempotent (saving and refreshing the page changes the HTML, which means saving again produces an altered HTML) is worrisome. Disabling validation means we're hiding the issue, not solving it. It means we don't care that the HTML is changed on a second save.